controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -79,21 +80,19 @@ public void triggerFromCompletion(JobReport report) {
         try (Lock lock = applications().lock(report.applicationId())) {
             LockedApplication application = applications().require(report.applicationId(), lock);
             application = application.withJobCompletion(report, clock.instant(), controller);
-            
+
             // Handle successful starting and ending
             if (report.success()) {
-                if (order.givesApplicationChange(report.jobType())) {
+                if (order.givesNewRevision(report.jobType())) {
                     if (acceptNewRevisionNow(application)) {
-                        // Set this as the change we are doing, unless we are already pushing a platform change
-                        if ( ! ( application.deploying().isPresent() && 
-                                 (application.deploying().get() instanceof Change.VersionChange)))
-                            application = application.withDeploying(Optional.of(Change.ApplicationChange.unknown()));
```

Comment:
This can't be removed, then.

Reply:
I thought this test was redundant, since we were only supposed to accept one kind of change at a time. We most certainly should change Change!